spikingjelly/activation_based/ann2snn/recipes.py

```
@@ -0,0 +1,973 @@
+                    return data[key]
+            return next(iter(data.values()))
+        return data
+
+    def _check_mode(self):
+        err_msg = "You have used a non-defined VoltageScale Method."
+        if isinstance(self.mode, str):
+            if self.mode[-1] == "%":
+                try:
+                    float(self.mode[:-1])
+                except ValueError:
+                    raise NotImplementedError(err_msg)
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)